docs/plugin/messaging.md

```diff
@@ -0,0 +1,886 @@
+In the past, prior to the R3 version of Thunder, logging, tracing and warning reporting were completely separate issues and were handled individually. However, even though these message types have distinct characteristics, we recognized the advantages of consolidating them into a unified framework, which is now referred to as `Messaging`. In the early versions of R3, we still had tracing enabled by default and messaging was in the early development. It is the R4 version of Thunder where `Messaging` is finally the default framework to handle all types of messages.
+
+
+# Advantages of using messaging
+
+There are of course several advantages of using this framework to deliver the messages over simply using something like `printf`:
+
+* There are not only different types of messages, but also categories within these types, and each of them can be enabled/disabled at runtime
+* No serialization penalty between different processes
+* Always getting full lines one after another since all messages are timestamped
+* Possibility of redirecting the messages to another output like, e.g., a console, syslog, a file or a network stream, which is particularly useful in the case of containers, and this is something we are working towards
```

Comment:
> and this is something we're working towards

Can you be clear which bit of this bullet we're working towards and which actually exists?